code/modules/mob/mob_movement.dm

```diff
@@ -456,6 +456,13 @@
 	set name = "Move Upwards"
 	set category = "IC"
 
+	var/turf/current_turf = get_turf(src)
+	var/turf/above_turf = SSmapping.get_turf_above(current_turf)
+
+	if(can_zFall(above_turf, 1, current_turf, DOWN)) //Will be fall down if we go up?
+		to_chat(src, "<span class='notice'>You are not superman.<span>")

```

Comment:
Maybe something like "You jump on place." is better?
and show it to all near.

Reply:
i'd rather not, that's not very universal to all cases were this wouldnt work

Reply:
Does / should this account for creatures that can fly? It would be interesting to see.

Reply:
it does

Reply:
anything that would float above open space would be able to move up. The only cave-at is that you can move up even if there's nothing to move up fromwhen theres no gravity, but that is something we can work in later.

Reply:
procedural assignment of z-levels to allow for arbitrary heights of locations? 😳

Reply:
huh?

Reply:
I say about this check.

Reply:
shouldn't "superman" be capitalized? it's a proper noun